packages/eas-cli/src/update/updateUtils.ts

```diff
@@ -0,0 +1,129 @@
+
+type TruncatedUpdate = Pick<
+  Update,
+  'updateGroup' | 'updateMessage' | 'createdAt' | 'runtimeVersion'
+> & { platforms: string; actor: User; id?: string; platform?: string };
+
+const PAGE_LIMIT = 10_000;
+
+export async function viewUpdateReleaseAsync({
+  appId,
+  releaseName,
+}: {
+  appId: string;
+  releaseName: string;
+}): Promise<{
+  id: string;
+  releaseName: string;
+  updates: TruncatedUpdate[];
+}> {
+  const data = await withErrorHandlingAsync(
+    graphqlClient
+      .query<
+        {
+          app: {
+            byId: {
+              updateReleaseByReleaseName: {
+                id: string;
+                releaseName: string;
+                updates: TruncatedUpdate[];
+              };
+            };
+          };
+        },
+        {
+          appId: string;
+          releaseName: string;
+        }
+      >(
+        gql`
+          query ViewRelease($appId: String!, $releaseName: String!) {
+            app {
+              byId(appId: $appId) {
+                updateReleaseByReleaseName(releaseName: $releaseName) {
+                  id
+                  releaseName
+                  updates(offset: 0, limit: ${PAGE_LIMIT}) {
+                    id
+                    updateGroup
+                    updateMessage
+                    createdAt
+                    platform
+                    runtimeVersion
+                    actor {
+                      ... on User {
+                        username
+                      }
+                      ... on Robot {
+                        firstName
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        `,
+        {
+          appId,
+          releaseName,
+        }
+      )
+      .toPromise()
+  );
+
+  return data.app.byId.updateReleaseByReleaseName;
+}
+
+export async function getUpdates(options: {
+  projectId: string;
+  releaseName: string;
+  platformFlag?: string;
+  allFlag?: boolean;
+}) {
+  const { projectId, releaseName, platformFlag, allFlag = false } = options;
+
+  const UpdateRelease = await viewUpdateReleaseAsync({
+    appId: projectId,
+    releaseName,
+  });
+
+  const filteredUpdates = UpdateRelease.updates.filter(update => {
+    if (platformFlag === undefined) {
+      return update;
+    }
+
+    return platformFlag.split(',').includes(update.platform ?? '');
+  });
+
+  if (allFlag) {
+    return filteredUpdates;
+  }
+
+  const updatesByGroup = filteredUpdates.reduce(
+    (acc, update) => {
+      const { id, platform, ...rest } = update;
+      const platforms = [...([acc[update.updateGroup]?.platforms] ?? []), platform]
+        .filter(Boolean)
+        .sort()
+        .join(', ');
+
+      return {
+        ...acc,
+        [update.updateGroup]: {
+          ...rest,
+          platforms,
+        },
+      };
+    },
+    {} as {
+      [i: string]: TruncatedUpdate;
+    }
+  );
```

Comment:
I'm a bit worried giving the option to also return 'update groups' will confuse people.

I think we could simplify this command by making it "well typed" and only return `updates`?

Is it even possible that we want to make this into a `view update group` command? The flow could be that a user sees an `update group` in the `release:view` command and wants a more granular view?

Like, what is the use case for see a list of all of the updates instead of update groups?

Reply:
> I think we could simplify this command by making it "well typed" and only return updates?
What do you mean by well typed? Can you provide an example?

> Is it even possible that we want to make this into a view update group command? The flow could be that a user sees an update group in the release:view command and wants a more granular view?

We do have the `update:show` command, which is not yet implemented. You're right that this command is nearly identical to the `release:view` command. The difference is that you can filter by platform and it does not print details about the release. I'm happy to remove this command if it's not needed, btw. @fson do you have an opinion about having `release:view` and `update:list`, which both require the developer to provide a release name and both print out a list of update groups?

> Like, what is the use case for see a list of all of the updates instead of update groups?

If people want to directly edit platform-specific updates, this would be needed. With that, I removed the `--all` flag from this PR because developers can use our API to do this work, or we can add it later. I'm not able to think of common flows that would use that information, and taking it out makes our CLI cleaner.

Reply:
This is a good question, would love to hear what @fson thinks about it.